src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -64,12 +64,14 @@ public class Itinerary implements ItinerarySortKey {
   /* other properties */
 
   private final List<SystemNotice> systemNotices = new ArrayList<>();
+  private final boolean isSearchWindowAware;
   private List<Leg> legs;
 
   private ItineraryFares fare = ItineraryFares.empty();
 
-  public Itinerary(List<Leg> legs) {
+  public Itinerary(List<Leg> legs, boolean isSearchWindowAware) {
```

Comment:
The boolean argument is hard to guess/read. One way to get around it, is to use factory methods:

Make the constructor private and create two factory methods, maybe: `Itinerary.createScheduledTransitItinerary(legs)` and `Itinerary.createDirectItinerary(legs)`. This will hide the flag  `searchWindowAware` from the producers (they do not care) and make it an itinerary thing. The coupling between the producer and the flag also become clear and safe - there is less room for setting it wrong. If you want you can create a separate factory method for flex e.g. `createTransitItinerary`, `createDirectFlexItinerary` and `createDirectStreetItinerary`. The last two would have the same implementation, but they are easier to use - less room for mistakes.

Reply:
I've implemented this in 0538587eec7906c208045f5f4a02a7a2860eef92.

The last suggestion of using separate methods for flex and street is hard without a larger refactoring because the two routers (A* and flex) use the same class to map from their internal representation to itineraries. I don't think it's worth it - do you, @t2gran?